src/scripting/lua_common.cpp

```diff
@@ -745,8 +745,9 @@ bool luaW_toconfig(lua_State *L, int index, config &cfg)
 	// Then convert the attributes (string indices).
 	for (lua_pushnil(L); lua_next(L, index); lua_pop(L, 1))
 	{
-		if (lua_isnumber(L, -2)) continue;
-		if (!lua_isstring(L, -2)) return_misformed();
+		int indextype = lua_type(L, -2);
+		if (indextype == LUA_TNUMBER) continue;
+		if (indextype != LUA_TSTRING) return_misformed();
```

Comment:
I don't have a problem with this, but I do question its necessity? From what I understood, it was `lua_tostring` and `lua_tonumber` that modified the stack value, but not `lua_isstring` or `lua_isnumber`.

That said, I suppose the old way would have skipped over tags such as `[12345]`, which are technically legal WML (even though no-one actually uses such tags that I know of).

Reply:
experimentation shows lua_isstring and lua_isnumber do the conversion to be able to tell is succeeded.

once bitten twice shy. If we had a problem with it one place, we probably just have not noticed it elsewhere.

Reply:
Huh, that's pretty unexpected (and frankly kind of worrying, for a test to be mutating the data its testing), but alright.

Reply:
As far as I can see, the only place Lua cares about the fact that data can be mutated is `lua_next` (in C++) and the `next(table, index)` Lua method.

One could argue the error is in the implementation of `lua_next` since the remainder of the language takes pains to ensure it does not matter. To be honest, I was amazed when I saw the constraints; I could understand them, once I saw them, but I expected Lua to be more resilient in this.